Week1 [STEP 2] 고도

gxdxt opened the pull request:
@DragonTigerXYZ 

안녕하세요 `앨런(Alan)` ! 

STEP 2 에 대한 PR 입니다!

for loop에 대한 예제 후,
 아이스크림 그리기 함수를 구현하였습니다!

인자로 아이스크림의 너비 / 높이 그리고 막대의 높이 값을 받아서 print하는 func을 구현하였습니다.
아이스크림을 그려줄 때에, 두 가지 선택지가 있었습니다.
***
1️⃣ 한 줄에 아이스크림 모양을 한 번에 찍기 (ex. icecreamPart = "*********")
2️⃣ 한 줄에 한 칸씩 아이스크림 모양을 찍기
***
결국 이 선택지들 중 `2번`을 선택하였습니다. 물론 빠른 실행을 위해서는 한 번에 한줄의 아이스크림 모양을 찍어내는 것이 좋겠지만, 다양한 모양의 아이스크림을 그리는 데에는 2번을 선택하는 것이 낫겠다고 판단하였습니다.
즉, 아래 코드처럼 값을 하나씩 찍도록 구현하는 것이 후에 아이스크림의 종류(~~혹은 모양~~)이 변경될 때에 `변경이 용이할 것`이라 판단하여 아래 코드를 선택하여 구현하였습니다.
```swift
for _ in 0...widthOfIceCream-1{
            icecreamPart += "*"
        }
```

그 이후 고민을 했던 부분은 바로 막대를 그려주는 부분이었습니다.
막대를 그려주는 부분은 항상 아이스크림의 가운데에 위치해야 하기에, 가변적인 아이스크림 너비에 따라서 해당 값을 부여해야 했습니다.
그렇기에 `해당 값(아이스크림 너비)을 2로 나눈 뒤, +1 그리고 -1 한 값을 만났을 때에 막대 모양을 넣어주도록 ` 코드 구현을 했습니다.
이 부분 역시, 막대 모양도 언제든 변경할 수 있도록 하나씩 String을 추가한 후 마지막에 그려주는 방향으로 코드를 작성했습니다.
물론 배운 것처럼 print(_:seperator:terminator)를 통해서 한 번에 찍어낼 수 있지만, 위와 같은 이유로 하나 씩 String을 잇는 코드로 구현하였습니다.
```swift
//the stick is placed in the middle of the width of the ice cream
            if ((j == widthOfIceCream/2 - 1) || (j == widthOfIceCream/2 + 1)) {
                stickPart += "|"
            } else {
                stickPart += " "
            }
```

🌲다만, (아이스크림의) 너비가 홀수일 떄에는 정상적으로 나오지만, 짝수일 때에는 제가 작성한 코드가  막대를 정 가운데가 아닌 옆으로 출력하였습니다. 해당 부분에 대한 아이디어 그리고 제 코드의 문제점이 있다면 가감없이 __매운맛으로__ 리뷰 부탁드립니다! 
항상 감사한 마음으로 리뷰 받겠습니다! 😄 

***

잘못 PR하여 다시 보냅니다 😢

Commit:
Create the 'for loop' and the function called 'printMuYaHo'

Commit:
Practice print(_:separator:terminator:)

Commit:
Create the func for drawing an Ice Cream and Add the flowchart

DragonTigerXYZ commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -15,6 +15,9 @@ for count in 1...6 {
     print(num, "번째 무야호!")
+    
 }
 
```
Step2에 불필요한 코드로 보이네요👀

DragonTigerXYZ commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -10,3 +10,11 @@ import Foundation
 
 print("Hello, World!")
 
```
Hello `고도 님` World~ 👀

불필요한 코드 같습니다~

DragonTigerXYZ commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -21,3 +22,34 @@ func printMuYaHo(num: Int) {
     
 }
 
+func drawAnIceCream(widthOfIceCream: Int, heightOfIceCream: Int, heightOfStick: Int) {
```
메서드명이 좀 길게 늘어지는 느낌이에요. 
어떻게 바꿔보는게 좋을까요?

gxdxt replied:
조사는 생략해도 될 것 같지만 (`drawIceCream`), 그 이상으로 메소드 명을 축약하면 의미 전달이 잘 되지 않을 것 같습니다. `drawIC` 같이 짧게는 어렵지 않을까요! 🤔

DragonTigerXYZ replied:
@yoosoony 

Naming에는 보통 관사는 생략합니다.
`Swift API Design Guidelines`을 한번 찾아볼까요?✍️

DragonTigerXYZ commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -21,3 +22,34 @@ func printMuYaHo(num: Int) {
 }
 
+func drawAnIceCream(widthOfIceCream: Int, heightOfIceCream: Int, heightOfStick: Int) {
+    var icecreamPart: String = ""
+    var stickPart: String = ""
+    
```
변수의 데이터 타입에는 어떤게 있고 
`고도 님`께서 왜 var를 선택하셨을까요?

gxdxt replied:
변수 데이터 타입은 var과 let이 대표적인데, var은 코드가 전개되며 변하는 값에, let은 한 번 선언된 이후로 변경되지 않는 값에 설정합니다! 제 코드같은 경우에는 해당 변수가 `지속적으로 UPDATE되어`야 하기 때문에 `var`을 선택했습니다!

Commit:
feat : Add the func that draws Pepero

Commit:
Merge branch 'first_br' into main

Commit:
chore : Delete the file for drawing Pepero
 - The file for drawing Pepero was deleted because it isn't for STEP 2
 - The codes that print the count was deleted because it isn't for STEP 2

Commit:
docs : Add a folder for managing the flowchart
 chore : Simplify the flowchart
